packages/jest-runner/src/run_test.js

```diff
@@ -100,8 +99,16 @@ export default function runTest(
     mapCoverage: globalConfig.mapCoverage,
   });
   const start = Date.now();
-  return testFramework(globalConfig, config, environment, runtime, path)
-    .then((result: TestResult) => {
+  return environment.setup().then(async testSetup => {
+    setGlobal(environment.global, 'setup', testSetup);
```

Comment:
This shouldn't be here, we don't want to expose this function on the global object.

Reply:
For some use case it's necessary to set global variables in the setup function for later usage.(i.e, user can store puppeteer's browser instance for all test cases instead of launching bunch of new ones)

Reply:
maybe
```javascript
return environment.setup().then(async setupGlobal => {
    setGlobal(environment.global, 'setup', setupGlobal);
```
would be better

Reply:
I prefer not to do this. The change in this PR should not automatically add things to the global scope. The setup function in the environment has access to the global scope and we can leave it to the responsibility of the user.